db/snapshot/backup_task.hh

```diff
@@ -20,23 +20,23 @@ namespace snapshot {
 
 class backup_task_impl : public tasks::task_manager::task::impl {
     snapshot_ctl& _snap_ctl;
-    shared_ptr<s3::client> _client;
+    sstring _endpoint;
     sstring _bucket;
     sstring _prefix;
     std::filesystem::path _snapshot_dir;
+    std::vector<s3::upload_progress> _progress_per_shard{smp::count};
     bool _remove_on_uploaded;
-    s3::upload_progress _progress = {};
 
     future<> do_backup();
-    future<> upload_component(sstring name);
+    future<> upload_component(shared_ptr<s3::client>& client, abort_source& as, s3::upload_progress& progress, sstring name);
 
 protected:
     virtual future<> run() override;
 
 public:
     backup_task_impl(tasks::task_manager::module_ptr module,
                      snapshot_ctl& ctl,
-                     shared_ptr<s3::client> cln,
+                     sstring endpoint,
```

Comment:
I'd appreciate if the change that makes backup_task_impl construct with string endpoint rather than shared-ptr client, was made as a separate preparational patch

Reply:
Done